spynnaker/pyNN/models/neuron/synaptic_manager.py

```diff
@@ -461,7 +461,10 @@ def _get_ring_buffer_to_input_left_shifts(
                     spikes_per_second = self._spikes_per_second
                     if isinstance(app_edge.pre_vertex,
                                   SpikeSourcePoissonVertex):
-                        spikes_per_second = app_edge.pre_vertex.rate
+                        rate = app_edge.pre_vertex.rate
+                        # If non-zero rate then use it; otherwise keep default
+                        if (sum(rate) != 0):
```

Comment:
Almost - you need some of what you had before, as you can only sum things that are lists (unless this is always guaranteed to be a list).

Reply:
It's guaranteed to be a list (SpikeSourcePoissonVertex.rate is set using utility_calls.convert_param_to_numpy, which enforces that single values become single-valued arrays).